pkgs/mime/lib/src/magic_numbers.dart

```diff
@@ -0,0 +1,154 @@
+  MagicNumber('image/jpeg', hex('FFD8')),
+  MagicNumber('image/png', hex('89504E470D0A1A0A')),
+  MagicNumber('image/tiff', hex('49492A00')),
+  MagicNumber('image/tiff', hex('4D4D002A')),
+  MagicNumber('audio/aac', hex('FFF1')),
+  MagicNumber('audio/aac', hex('FFF9')),
+  MagicNumber('audio/weba', hex('1A45DFA3')),
+  MagicNumber('audio/mpeg', hex('494433')),
+  MagicNumber('audio/mpeg', hex('FFFB')),
+  MagicNumber('audio/ogg', hex('4F7075')),
+  MagicNumber(
+    'video/3gpp',
+    hex('000000006674797033677035'),
+    mask: hex('FFFFFF00FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797061766331'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797069736F32'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797069736F6D'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('00000000667479706D703431'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('00000000667479706D703432'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber('model/gltf-binary', hex('46546C67')),
+
+  /// The WebP file format is based on the RIFF document format.
+  /// -> 4 bytes have the ASCII characters 'R' 'I' 'F' 'F'.
+  /// -> 4 bytes indicating the size of the file
+  /// -> 4 bytes have the ASCII characters 'W' 'E' 'B' 'P'.
+  /// https://developers.google.com/speed/webp/docs/riff_container
+  MagicNumber(
+    'image/webp',
+    hex('524946460000000057454250'),
+    mask: hex('FFFFFFFF00000000FFFFFFFF'),
+  ),
+
+  MagicNumber('font/woff2', hex('774f4632')),
+
+  /// High Efficiency Image File Format (ISO/IEC 23008-12).
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'h' 'e' 'i' 'c'.
+  /// https://www.iana.org/assignments/media-types/image/heic
+  MagicNumber(
+    'image/heic',
+    hex('000000006674797068656963'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'h' 'e' 'i' 'x'.
+  MagicNumber(
+    'image/heic',
+    hex('000000006674797068656978'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'm' 'i' 'f' '1'.
+  MagicNumber(
+    'image/heif',
+    hex('00000000667479706D696631'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+];
+
+Uint8List hex(String encoded) {
+  final result = Uint8List(encoded.length ~/ 2);
+  for (var i = 0; i < result.length; i++) {
+    final offset = i * 2;
+    result[i] = int.parse(encoded.substring(offset, offset + 2), radix: 16);
```

Comment:
I think we have a hex codec somewhere.
... found it: https://pub.dev/documentation/convert/latest/convert/HexCodec-class.html

Don't know if it's worth using. This code can definitely be optimized.